core/src/main/java/io/grpc/internal/RetriableStream.java

```diff
@@ -603,5 +602,7 @@ State committed(Substream winningSubstream) {
 
     // GuardedBy RetriableStream.lock
     boolean closed;
+
+    int previousAttempts;
```

Comment:
Shouldn't this be incremented somewhere? It looks like it's always `0`?

Reply:
It is incremented in `retry(substream.previousAttempts + 1)`. (For hedging, this number in each of the hedged substreams will be set incrementally when RetriableStream starts.) The field `previousAttempts` in `Substream` is effectively final through.

Reply:
FWIW, I don't think the intent was for hedged RPCs to set the `grpc-previous-rpc-attempts` header.